botocore/utils.py

```diff
@@ -632,6 +632,15 @@ def check_dns_name(bucket_name):
     return True
 
 
+def switch_to_s3_sigv2_presigner(signature_version, signing_name, **kwargs):
```

Comment:
That's not strictly true either. I only return if I think it should be switched.